### DISCUSSION TITLE
GitHub Action provide runners that arm64 and Linux?

### DISCUSSION THREAD

**Discussion by lgu-ampere** [community/community#142209] [Actions] [answered]
### Select Topic Area

Question

### Body

Does GitHub Action provide runners that arm64 and Linux? I didn't find the workflow label for it. Just find arm64 for macOS

**Comment by Rhys-T** [comment]
Yes, but they're not available on the free tier yet. GitHub says they're planning on opening them up to everyone by the end of 2024. See #19197 for more info.

You may be able to work around it for now by installing user-mode QEMU onto an x86-64 Linux runner to let it run ARM64 executables, but it's going to be significantly slower.

**Reply by Rhys-T** [reply, marked as answer]
Update: It looks like GitHub might have postponed opening arm64-linux runners to everyone until early 2025. See:
- https://github.com/orgs/community/discussions/19197#discussioncomment-11093292
- https://github.blog/news-insights/product-news/universe-2024-previews-releases/#more-compute-resources-and-better-insights-for-your-workflows (open the 'read more' section)

**Reply by samus-aran** [reply]
@lgu-ampere ☝️ @Rhys-T did a great answer and providing the latest updates , I would suggest marking it as the solution 😉

**Reply by Rhys-T** [reply]
I was originally installing QEMU by adding a step that does `sudo apt install qemu-user-static`, but I started getting errors for that at one point. Changing it to `sudo apt update && sudo apt install qemu-user-static` seemed to fix it, but I finally ended up replacing all that with <https://github.com/docker/setup-qemu-action>, which seems to install faster.
